### PR TITLE
BD-20 feat: social auth 

### PR DESCRIPTION
**Pull request by SlavaMryasov**

### CODE REVIEW

File: components/socialAuth/GitHabAuth.tsx
Comment (sfgardev) [inline components/socialAuth/GitHabAuth.tsx:1]: the component should be named GithubAuth
image

File: components/socialAuth/GitHabAuth.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
+
+import { Icon } from 'uikit-inctagram'
+
+export const GitHabAuth = () => {
```
Comment (sfgardev) [inline components/socialAuth/GitHabAuth.tsx:5]: rename component to GithubAuth

File: components/socialAuth/GitHabAuth.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  const gitHubLogin = () => {
```
Comment (sfgardev) [inline components/socialAuth/GitHabAuth.tsx:6]: rename to githubLogin

File: components/socialAuth/GoogleAuth.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import { authActions } from '@/features'
+import { authApi } from '@/services'
+import { useAppDispatch } from '@/store'
+import { useGoogleLogin } from '@react-oauth/google'
+import { useRouter } from 'next/router'
+import { Icon } from 'uikit-inctagram'
+
+export const GoogleAuth = () => {
+  const [google] = authApi.useGoogleMutation()
```
Comment (sfgardev) [inline components/socialAuth/GoogleAuth.tsx:9]: rename google to googleLogin

File: components/socialAuth/GoogleAuth.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      try {
+        const { code } = res
+
+        const response = await google({ code }).unwrap()
+
+        dispatch(authActions.login({ accessToken: response.accessToken }))
+
+        const email = response.email
+
+        router.push(`/profile/${email}`)
```
Comment (sfgardev) [inline components/socialAuth/GoogleAuth.tsx:29]: should be profile id, not email
you can take id from decoded token, like you have done on github page

File: pages/github/index.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  useEffect(() => {
+    if (accessToken) {
+      const getUserIdFromToken = (token: string) => {
+        try {
+          const payload = token.split('.')[1]
+          const decodedPayload = JSON.parse(atob(payload))
+
+          return decodedPayload.userId
+        } catch (err) {
+          console.error('Ошибка извлечения userId из токена', err)
```
Comment (sfgardev) [inline pages/github/index.tsx:22]: english please

File: pages/github/index.tsx [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+import { Spinner, getAuthLayout } from '@/components'
+import { authActions } from '@/features'
+import { useAppDispatch } from '@/store'
+import { useRouter } from 'next/router'
+
+const OAuthGitHub = () => {
+  const router = useRouter()
+  const dispatch = useAppDispatch()
+  const { accessToken, email } = router.query
```
Comment (sfgardev) [inline pages/github/index.tsx:11]: do like this
const accessToken = router.query.accessToken as string
and no need to do type assertion in two places
image

and why do we need email here?

File: pages/github/index.tsx [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        }
+      }
+      const userId = getUserIdFromToken(accessToken as string)
+
+      if (userId) {
+        dispatch(authActions.login({ accessToken: accessToken as string }))
+        router.push(`/profile/${userId}`)
+      }
+    }
+  }, [accessToken, email, router, dispatch])
```
Comment (sfgardev) [inline pages/github/index.tsx:34]: you do not use email in useEffect, why is it in dependency array?

File: services/auth/authSlice.ts [hunk 8]
```diff
@@ -20,6 +22,9 @@ export const authApi = createApi({
     confirmRegistration: builder.mutation<{}, ConfirmRegistrationRequest>({
       query: body => ({ body, method: 'POST', url: 'auth/registration-confirmation' }),
     }),
+    google: builder.mutation<GoogleResponse, GoogleRequest>({
```
Comment (sfgardev) [inline services/auth/authSlice.ts:25]: googleLogin should be better name

File: services/auth/types.ts [hunk 9]
```diff
@@ -24,6 +24,15 @@ export type ResendConfirmationCodeRequest = {
   baseUrl: string
   email: string
 }
+export type GoogleRequest = {
```
Comment (sfgardev) [inline services/auth/types.ts:27]: GoogleLoginRequest

File: services/auth/types.ts [hunk 10]
```diff
@@ -24,6 +24,15 @@ export type ResendConfirmationCodeRequest = {
...
+  code: string
+  scope?: string
```
Comment (sfgardev) [inline services/auth/types.ts:29]: what is scope?
image

File: services/auth/types.ts [hunk 11]
```diff
@@ -24,6 +24,15 @@ export type ResendConfirmationCodeRequest = {
...
+}
+
+export type GoogleResponse = {
```
Comment (sfgardev) [inline services/auth/types.ts:32]: GoogleLoginResponse

File: pages/sign-in/index.tsx [hunk 12]
```diff
@@ -1,16 +1,16 @@
 import { SignInForm, getAuthLayout } from '@/components'
+import { GitHabAuth, GoogleAuth } from '@/components/socialAuth'
```
Comment (sfgardev) [inline pages/sign-in/index.tsx:2]: you can import from @/components

**Review by sfgardev — CHANGES_REQUESTED** [review]

File: components/socialAuth/GitHabAuth.tsx
Comment (SlavaMryasov) [reply components/socialAuth/GitHabAuth.tsx:1]: done

File: components/socialAuth/GitHabAuth.tsx (hunk 1 above)
Comment (SlavaMryasov) [reply components/socialAuth/GitHabAuth.tsx:5]: done

File: components/socialAuth/GitHabAuth.tsx (hunk 2 above)
Comment (SlavaMryasov) [reply components/socialAuth/GitHabAuth.tsx:6]: GithubLogin is component. Сomponents must be named in capital letters

File: components/socialAuth/GoogleAuth.tsx (hunk 3 above)
Comment (SlavaMryasov) [reply components/socialAuth/GoogleAuth.tsx:9]: done

File: components/socialAuth/GoogleAuth.tsx (hunk 4 above)
Comment (SlavaMryasov) [reply components/socialAuth/GoogleAuth.tsx:29]: done

File: pages/github/index.tsx (hunk 5 above)
Comment (SlavaMryasov) [reply pages/github/index.tsx:22]: done

File: pages/github/index.tsx (hunk 6 above)
Comment (SlavaMryasov) [reply pages/github/index.tsx:11]: done

File: pages/github/index.tsx (hunk 7 above)
Comment (SlavaMryasov) [reply pages/github/index.tsx:34]: done

File: services/auth/authSlice.ts (hunk 8 above)
Comment (SlavaMryasov) [reply services/auth/authSlice.ts:25]: done

File: services/auth/types.ts (hunk 9 above)
Comment (SlavaMryasov) [reply services/auth/types.ts:27]: done

File: services/auth/types.ts (hunk 10 above)
Comment (SlavaMryasov) [reply services/auth/types.ts:29]: done

File: services/auth/types.ts (hunk 11 above)
Comment (SlavaMryasov) [reply services/auth/types.ts:32]: done

**Review by sfgardev — APPROVED** [review]

**Pull request merged**